tests/io_components/test_metapartition.py

```diff
@@ -1627,3 +1627,16 @@ def test_parse_nested_input_schema_compatible_but_different():
     mp = parse_input_to_metapartition(df_input, metadata_version=4)
     expected_schema = make_meta(pd.DataFrame({"A": ["str"]}), origin="expected")
     assert mp.table_meta["table"] == expected_schema
+
+
+def test_parse_input_schema_formats():
+    df = pd.DataFrame({"B": [pd.Timestamp("2019")]})
+    formats_obj = [
```

Comment:
Nitpick: you could use `@pytest.mark.parametrize` here

Reply:
Nvm. I see that you're re-using the dataframe

Reply:
Yes, that's why I stuck with the `for` loop here.